### ISSUE TITLE
Constant iou=0.7, and a very high average loss when training one class

### ISSUE THREAD

**Issue by abdallah1097** [AlexeyAB/darknet#7246]
If you have an issue with training - no-detections / Nan avg-loss / low accuracy:
    * what command do you use? !./darknet detector train /content/darknet/build/darknet/x64/data/obj.data /content/darknet/cfg/yolov4-custom.cfg /content/darknet/yolov4.conv.137 -dont_show -map -gpus 0 
    * what dataset do you use? My own dataset, but create labels then used OIDv4_ToolKit to convert_annotations and created train.txt and edited yolo-custom.cfg, obj.data and obj.names
    * check your dataset - run training with flag `-show_imgs` i.e. `./darknet detector train ... -show_imgs` and look at the `aug_...jpg` images, do you see correct truth bounded boxes? No, the augmented ruins the data! How to disable augmentation?
    * rename your cfg-file to txt-file and drag-n-drop (attach) to your message here
[yolo-obj.txt](https://github.com/AlexeyAB/darknet/files/5811186/yolo-obj.txt)
    * show such screenshot with info

I'm training Darknet Yolo with 1 class (Have 9000 training examples!), but I have this sample of the output:
```bash
v3 (iou loss, Normalizer: (iou: 0.07, obj: 1.00, cls: 1.00) Region 150 Avg (IOU: 0.000000), count: 1, class_loss = 683.035156, iou_loss = 0.000000, total_loss = 683.035156 
```
The iou remains **constant** at 0.07 and a high-class loss. And most awfully I get is:
```bash
(next mAP calculation at 1000 iterations) 
 77: 1497.594727, 1736.966309 avg loss, 0.000000 rate, 5.477014 seconds, 4928 images, 3.481637 hours left
```
What is the problem that causes this **constant** high average loss over time? Since I'm using a pre-trained backend it must converge, or overfit!

**Details**

The most relevant parts of the *yolov4-custom.cfg* file:
```bash
batch=64
subdivisions=16
width=512
height=512
channels=1
momentum=0.949
decay=0.0005

max_batches = 2000
steps=1600,1800

...

filters=18
activation=linear


[yolo]
mask = 6,7,8
anchors = 12, 16, 19, 36, 40, 28, 36, 75, 76, 55, 72, 146, 142, 110, 192, 243, 459, 401
classes=1

```

The *obj.data* file:
```bash
classes = 1
train  = /content/darknet/build/darknet/x64/data/train.txt
names = /content/darknet/build/darknet/x64/data/obj.names
backup = /content/darknet/build/darknet/x64/backup/
```
The *obj.name* file:
```bash
Object
```

**Comment by nizarmasmoudi** [comment]
Make sure your images' shapes fit the network. Your network size is `512 x 512` which means that your images have to be less or equal than `(512, 512, 3)` in terms of shape. Otherwise your network will only process a part of the image and won't find the objects you are telling it to look for. Either increase the network size or split your images if they are too large and you have computing limitations.

You should re-calculate anchors as well if your objects are small.

I hope this helped.

**Comment by stephanecharette** [comment]
> Otherwise your network will only process a part of the image and won't find the objects you are telling it to look for.

That is not true.  It still processes the entire image, no part is cropped out.  But your images will be stretched to match the network dimensions.  If the training images and the images you target are both the same dimensions, then stretching doesn't matter since they both will experience the exact same style/type of distortions.  If your training images are for example 512x512 but the images you use for inference are 1280x768, this will cause you issues, especially as the objects get smaller.

**Comment by nizarmasmoudi** [comment]
So, if I got this right, it doesn't affect accuracy if you train a `416 x 416` network with `1080 x 1080` images. I had the same problem he had and I fixed it by splitting my images to fit the network size. I thought that was the issue but now I'm confused.

**Comment by Srikeshram** [comment]
@abdallah1097 Did you solve your issue?
